$NetBSD: patch-ad,v 1.1 2006/11/03 07:53:55 rillig Exp $

unzipsfx does not need libz, so a special compile rule is needed for it.

--- unix/Makefile.orig	Mon Oct 30 22:45:35 2006
+++ unix/Makefile	Mon Oct 30 22:48:02 2006
@@ -87,7 +87,7 @@
 OBJS = $(OBJS1) $(OBJS2) $(OBJS3) $M$O
 LOBJS = $(OBJS)
 OBJSDLL = $(OBJS:.o=.pic.o) api.pic.o
-OBJX = unzipsfx$O $(CRC32)$O crctab_$O crypt_$O extract_$O fileio_$O \
+OBJX = unzipsfx$O crc32_$O crctab_$O crypt_$O extract_$O fileio_$O \
 	globals_$O inflate_$O match_$O process_$O ttyio_$O $M_$O
 LOBJX = $(OBJX)
 OBJF = funzip$O $(CRC32)$O cryptf$O globalsf$O inflatef$O ttyiof$O
@@ -306,6 +306,11 @@
 	-$(CP) unzip.c unzipsfx.c
 	$(CC) -c $(CF) -DSFX unzipsfx.c
 	$(RM) unzipsfx.c
+
+crc32_$O:	crc32.c $(UNZIP_H) zip.h			# unzipsfx only
+	-$(CP) crc32.c crc32_.c
+	$(CC) -c $(CF) -DSFX crc32_.c
+	$(RM) crc32_.c
 
 crctab_$O:	crctab.c $(UNZIP_H) zip.h
 	-$(CP) crctab.c crctab_.c
